### PR TITLE
feat: Add stats for collections

### PR DESCRIPTION
**Pull request by Matehoo**
TODO:
- [x] proper testing

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes [#10](https://github.com/kodadot/loligo/issues/10)
- [ ] This PR is required for kodadot/nft-gallery#<issue_number>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've run the indexer and it hasn't failed
- [x] I've changed schema and performed a migration
- [ ] I found edge cases

### Screenshot

- [ ] My contribution is a **resolver** so I have pic from playground

### CODE REVIEW

File: src/mappings/shared/list.ts [hunk 1]
```diff
@@ -32,9 +32,13 @@ export async function list(context: Context) {
     isPositiveOrElseError(price)
     nft.price = price
     nft.updatedAt = timestamp
+    if (nft.collection.floor === 0n || nft.price < nft.collection.floor) {
+      nft.collection.floor = nft.price
+    }
 
-    success(OPERATION, `${nft.id} from ${caller}`)
     await context.store.save(nft)
+    await context.store.save(nft.collection)
```
Comment (vikiival) [inline src/mappings/shared/list.ts:40]: You dont need to do this imo, it's cascaded

File: src/mappings/utils/helper.ts [hunk 2]
```diff
@@ -66,3 +68,28 @@ export const isValidAddressPolkadotAddress = (address: string) => {
...
+  collection: CollectionEntity,
+  newOwner?: string,
+  originalOwner?: string
+): number {
+  const allUniqOwners: string[] = uniq(
+    compact([...map(collection.nfts, 'currentOwner').filter((owner) => owner !== originalOwner), newOwner])
+  )
+
+  return allUniqOwners.filter((owner) => owner !== collection.issuer).length
+}
```
Comment (vikiival) [inline src/mappings/utils/helper.ts:95]: I would rather do raw query like

```sql
Select COUNT(DISTINCT current_owner) as distribution, COUNT(current_owner) as count from nft_entity where collection_id = $1
```

**Review by vikiival — COMMENTED** [review]
Otherwise nice job

**Comment by vikiival** [comment]
WDYT @Matehoo ? 

I think it looks much cleaner

**Comment by Matehoo** [comment]
I also think this way is better, its certainly more performant than before

**Comment by vikiival** [comment]
>  I've run the indexer and it hasn't failed?

**Comment by Matehoo** [comment]
> I've run the indexer and it hasn't failed?

For test I ran the indexer for 30 mins and didnt get any errors related to the added functionality

**Pull request merged**